### ISSUE TITLE
Sending authenticated user data questions

### ISSUE THREAD

**Issue by CescoIV** [wdi-atx-12/project-01#2]
Hello, after authenticating a user and having them log in, we are unsure as to how to access that data from the user, and display the user's information. How do we route a successful user login with the user's json object?

repo: https://github.com/CescoIV/my_travels_log

**Comment by jhnsnc** [comment]
There are a few moving pieces:

- the session cookie (identifying an active user session)
- the auth strategy (for logging in a user or registering a new user)
- the database record (saved data about each user)

If you're using `passport` with a passport strategy plugin, like we did last Thursday, it will take care of a lot of it for you. Basically, when you setup your passport auth strategy, make sure you are creating a new User record in your database (or updating an existing one).

Then, assuming you set up passport to serialize/deserialize your user, you should be able to access `req.user` for any request made from a logged in user. The serialize bit is how passport encodes the active user into the session cookie (usually just providing an `_id`). The deserialize bit is how passort looks at an incoming request from an active/logged-in user and turns it into an object to store on `req.user`. That usually involves a simple database lookup given an `_id` (pulled from your serialized session cookie).

**TL;DR**
Use `passport` to manage all the tricky bits and just access user information for each request via `req.user`.
